src/builder/pod/mod.rs

```diff
@@ -452,12 +453,19 @@ impl PodBuilder {
         self
     }
 
-    pub fn termination_grace_period_seconds(
+    pub fn termination_grace_period(
         &mut self,
-        termination_grace_period_seconds: i64,
-    ) -> &mut Self {
+        termination_grace_period: &Duration,
+    ) -> OperatorResult<&mut Self> {
+        let termination_grace_period_seconds = termination_grace_period
+            .as_secs()
+            .try_into()
+            .map_err(|_| Error::DurationTooLong {
+                duration: *termination_grace_period,
+            })?;
```

Comment:
I fear this is a bigger thing. We have `Option<i32>` or `Option<i64>` multiple time in k8s-openapi.
E.g. the number of STS replicas is an i32 as well: https://github.com/stackabletech/hdfs-operator/blob/9830327fc493bca5f7d556ba34e04e6fa15ab587/rust/operator/src/hdfs_controller.rs#L699

Reply:
Yeah it'd be nice to resolve eventually but I wouldn't block on it.

Reply:
Yes. This is clearly a bigger issue, but fundamental issues lies within the OpenAPI spec, which doesn't define any *unsigned* integers, only signed ones (with different lengths). This can be fixed at different links in the chain:

- By K8s itself, by either specifying a `min` value of `0` or unsigned integers, by introducing a custom format or by using a JSON schema
- By k8s-openapi, which would manually fix known fields by using `u32` or `u64`. One example would be fields ending with `Seconds`.

Reply:
I added this issue to our backlog. We will tackle this in the future.